IDNA changes as discussed on the list (#88)
richsalz/draft-ietf-uta-rfc6125bis: draft-ietf-uta-rfc6125bis.md

   Additional qualifications apply (refer to the above-referenced
   specifications for details), but they are not relevant here.

2. An "internationalized domain name", i.e., a DNS domain name that
   includes at least one label containing appropriately encoded
   Unicode code points outside the traditional US-ASCII range.
   In particular, it contains at least one NC-label or P-label, but
   otherwise may contain any mixture of NR-LDH labels, P-labels,

>    or NC-labels. Refer to [[Section 7.3]] for further details and the rationale

Reviewer: ```suggestion
   or NC-labels. Refer to {{security-idn}} for further details and the rationale
```

I think that you intend to refer to this section, but I didn't check.

## Internationalized Domain Names {#security-idn}

The IETF document covering internationalized domain names is

> "IDNA2008" [IDNA-DEFS]. The Unicode Consortium publishes

Reviewer: ```suggestion
"IDNA2008" {{IDNA-DEFS}}. The Unicode Consortium publishes
```

(Brackets are less robust than the moustachios in kramdown-rfc.)

a similar document known as "UTS-46". To account for the use of UTS-46 in
practice, this document employs the terms "P-label" and "NC-label" as opposed
to "A-label" and "U-label", which are specific to IDNA2008.

As specified under {{verify}}, matching of internationalized domain names

> is performed on P-labels, not NC-labels.  As a result, potential confusion

Reviewer 2: This is the primary change of this PR, in my opinion.

Reviewer 3: Discussion is ongoing on the WG list. I would prefer to let that conversation play out completely before processing any PRs or related comments, in part because some folks there with relevant expertise are not GitHub users.

caused by the use of visually similar characters in domain names is likely
mitigated in certificate matching as described in this document.

UTS-46 allows names that are valid in IDNA2003 but not IDNA2008,
and additionally allows characters that are not valid in either IETF
document, such as emoji characters. This more lenient approach
carries additional risk of semantic ambiguity and additional security
considerations. ICANN recommends IDNA2008

> [https://features.icann.org/ssac-advisory-use-emoji-domain-names]

Reviewer: This needs a more concrete citation.

and correspondingly recommends against emoji characters in DNS names.
However, the internet contains old content published under IDNA2003,
and people enjoy emoji characters, so consumer applications often

> end up using the approach in [UTS-46]. DNS names that conform to

Reviewer: ```suggestion
end up using the approach in {{UTS-46}}. DNS names that conform to
```
